fog/ledger/server/src/key_image_router_service.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+use futures::{FutureExt, TryFutureExt};
+use grpcio::{DuplexSink, RequestStream, RpcContext};
+use mc_common::logger::{log, Logger};
+use mc_fog_api::{
+    ledger::{LedgerRequest, LedgerResponse},
+    ledger_grpc::{self, LedgerApi},
+};
+use mc_fog_ledger_enclave::LedgerEnclaveProxy;
+use mc_fog_uri::KeyImageStoreUri;
+use mc_util_grpc::rpc_logger;
+use mc_util_metrics::SVC_COUNTERS;
+
+use crate::router_handlers;
+
+#[derive(Clone)]
+pub struct KeyImageRouterService<E>
+where
+    E: LedgerEnclaveProxy,
+{
+    enclave: E,
+    shards: Arc<RwLock<HashMap<KeyImageStoreUri, Arc<ledger_grpc::KeyImageStoreApiClient>>>>,
+    query_retries: usize,
+    logger: Logger,
+}
+
+impl<E: LedgerEnclaveProxy> KeyImageRouterService<E> {
+    /// Creates a new LedgerRouterService that can be used by a gRPC server to
+    /// fulfill gRPC requests.
+    #[allow(dead_code)] // FIXME
```

Comment:
Do we still need this?

Reply:
This gets removed in the Server PR (#2921).